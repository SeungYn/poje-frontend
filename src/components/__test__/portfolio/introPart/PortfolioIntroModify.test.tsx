import PortfolioIntroModify from '@src/components/portfolio/introPart/PortfolioIntroModify';
import usePortfolioIntro from '@src/hooks/portfolio/intro/usePortfolioIntro';
import usePortfolioModifyForm from '@src/hooks/portfolio/intro/usePortfolioModifyForm';
import { mockPortfolioIntro } from '@src/test_util/mock/data/portfolio';
import { withAllContext } from '@src/test_util/util';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useRef } from 'react';

jest.mock('@src/hooks/portfolio/intro/usePortfolioIntro');
jest.mock('@src/hooks/portfolio/intro/usePortfolioModifyForm');

const mockUsePortfolioIntro: jest.Mock =
  usePortfolioIntro as unknown as jest.MockedFunction<any>;
const mockUsePortfolioModifyForm: jest.Mock =
  usePortfolioModifyForm as unknown as jest.MockedFunction<any>;

describe('PortfolioIntroModify Component test', () => {
  afterEach(() => {
    mockUsePortfolioIntro.mockReset();
    mockUsePortfolioModifyForm.mockReset();
  });

  it('snapshot test 로딩중 표시', () => {
    mockUsePortfolioIntro.mockImplementation(() => ({
      pfIntro: mockPortfolioIntro,
      isMutating: true,
      isFetching: true,
      updateIntro: () => {},
    }));
    mockUsePortfolioModifyForm.mockImplementation(() => ({
      copiedPfIntro: mockPortfolioIntro,
      onChangeInputEl: () => {},
      discriptionRef: useRef(),
      handleSubmit: () => {},
    }));

    const { asFragment } = render(withAllContext(<PortfolioIntroModify />));

    expect(screen.getByText('업데이트중')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('snapshot test 로딩중이 아닐 때 ❌', () => {
    mockUsePortfolioIntro.mockImplementation(() => ({
      pfIntro: mockPortfolioIntro,
      isMutating: false,
      isFetching: false,
      updateIntro: () => {},
    }));
    mockUsePortfolioModifyForm.mockImplementation(() => ({
      copiedPfIntro: mockPortfolioIntro,
      onChangeInputEl: () => {},
      discriptionRef: useRef(),
      handleSubmit: () => {},
    }));

    const { asFragment } = render(withAllContext(<PortfolioIntroModify />));

    expect(screen.queryByText('업데이트중')).not.toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('handleSubmit 호출 확인 ✅', async () => {
    const mockHandelSubmit = jest.fn((e) => e.preventDefault());
    mockUsePortfolioIntro.mockImplementation(() => ({
      pfIntro: mockPortfolioIntro,
      isMutating: false,
      isFetching: false,
      updateIntro: () => {},
    }));
    mockUsePortfolioModifyForm.mockImplementation(() => ({
      copiedPfIntro: mockPortfolioIntro,
      onChangeInputEl: () => {},
      discriptionRef: useRef(),
      handleSubmit: mockHandelSubmit,
    }));

    render(withAllContext(<PortfolioIntroModify />));

    const submitBtn = screen.getByText('확인');
    userEvent.click(submitBtn);

    expect(mockHandelSubmit).toBeCalled();
  });
});
