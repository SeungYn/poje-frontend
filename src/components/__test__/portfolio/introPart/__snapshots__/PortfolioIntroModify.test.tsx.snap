// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioIntroModify Component test snapshot test 로딩중 표시 1`] = `
<DocumentFragment>
  <form
    class="sc-ftTHYK sc-pyfCe jjiciW ghyGMw"
  >
    <div
      class="sc-bcXHqe csvyXT"
    >
      <div
        class="sc-gswNZR lnMQDM"
      />
      <p
        class="sc-dkrFOg dFrXTK"
      >
        업데이트중
      </p>
    </div>
    <input
      accept="image/gif, image/jpeg, image/png"
      name="file"
      style="display: none;"
      type="file"
    />
    <section
      class="sc-jrcTuL cDhclm"
    >
      <input
        class="sc-kDvujY sc-ipEyDJ fjINIp dPPJJF"
        name="title"
        type="text"
        value="제목을 입력해주세요."
      />
      <hr
        class="sc-csuSiG cUorDq"
      />
      <textarea
        class="sc-eDWCr fWpaGu"
        name="description"
      >
        내용을 입력해주세요.
      </textarea>
    </section>
    <article
      class="sc-hLBbgP hxWFmO"
    >
      <button
        class="sc-jSUZER sc-gKPRtg dCZjja"
        type="submit"
      >
        확인
      </button>
      <button
        class="sc-jSUZER sc-iBYQkv dCZjja bWOWdx"
      >
        취소
      </button>
    </article>
  </form>
</DocumentFragment>
`;

exports[`PortfolioIntroModify Component test snapshot test 로딩중이 아닐 때 ❌ 1`] = `
<DocumentFragment>
  <form
    class="sc-ftTHYK sc-pyfCe jjiciW ghyGMw"
  >
    <input
      accept="image/gif, image/jpeg, image/png"
      name="file"
      style="display: none;"
      type="file"
    />
    <section
      class="sc-jrcTuL cDhclm"
    >
      <input
        class="sc-kDvujY sc-ipEyDJ fjINIp dPPJJF"
        name="title"
        type="text"
        value="제목을 입력해주세요."
      />
      <hr
        class="sc-csuSiG cUorDq"
      />
      <textarea
        class="sc-eDWCr fWpaGu"
        name="description"
      >
        내용을 입력해주세요.
      </textarea>
    </section>
    <article
      class="sc-hLBbgP hxWFmO"
    >
      <button
        class="sc-jSUZER sc-gKPRtg dCZjja"
        type="submit"
      >
        확인
      </button>
      <button
        class="sc-jSUZER sc-iBYQkv dCZjja bWOWdx"
      >
        취소
      </button>
    </article>
  </form>
</DocumentFragment>
`;
